plugins/catalog-backend-module-gitlab/src/GitLabDiscoveryProcessor.ts

```diff
@@ -93,11 +93,21 @@ export class GitLabDiscoveryProcessor implements CatalogProcessor {
     const startTimestamp = Date.now();
     this.logger.debug(`Reading GitLab projects from ${location.target}`);
 
-    const projects = paginated(options => client.listProjects(options), {
+    const opts = {
       group,
-      last_activity_after: await this.updateLastActivity(),
       page: 1,
-    });
+    } as {
+      group: string;
+      page: number;
+      last_activity_after: string | undefined;
+    };
+
+    const lastActivity = (await this.cache.get(this.getCacheKey())) as string;
+    if (lastActivity !== '') {
```

Comment:
Mmm should probably rather be `if (lastActivity) {` to cover for undefined/null etc too.

Reply:
Won't it always be a string though, because of `as string;`?

Reply:
No. `as` is a typescript-level "theoretical" cast, it has absolutely no runtime effect (it's compiled away and in javascript there will be no trace of it left)

Reply:
We can get away with a lot less refactoring here as well right? I'd hope that sending `last_activity_after` as `undefined` has the same effect as not sending it at all.

Reply:
@freben made changes as per your recommendation. Tests still seem to pass.